frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+  selectedConstraints[0] = undefined;
+  selectedConstraints[1] = undefined;
+  $('#spConstraint' + tableNumber + tableColumnNumber).on('change', function () {
+    constraintId = $(this).attr('id');
+    idNum = parseInt($(this).attr('id').match(/\d+/g), 10);
+    constraints = document.getElementById(constraintId);
+    constraintsSelected = [];
+    for (var c = 0; c < constraints.length; c++) {
+      if (constraints.options[c].selected) constraintsSelected.push(constraints.options[c].value);
+    }
+    selectedConstraints[idNum] = constraintsSelected;
+  });
+}
+
+/**
+ * Function to save changes of spanner table
+ *
+ * @param {event} event event generated by clicking edit spanner button
+ * @return {null}
+ */
+const saveSpannerChanges = (event) => {
+  if (event.html() == 'Save Changes') {
+    showSnackbar('changes saved successfully !!', ' greenBg');
+  }
+
+  tableNumber = parseInt(event.attr('id').match(/\d+/), 10);
+  tableId = '#src-sp-table' + tableNumber + ' tr';
+  event.html("Edit Spanner Schema");
+  notPkArray = [];
+  initialColNameArray[tableNumber] = [];
+  currentPks = schemaConversionObj.SpSchema[src_table_name[tableNumber]].Pks;
+  updatedColsData = {
+    'UpdateCols': {
+    }
+  }
+  $(tableId).each(function (index) {
+    if (index > 1) {
+      tableName = src_table_name[tableNumber];
+      tableColumnNumber = parseInt($(this).find('.srcColumn').attr('id').match(/\d+/), 10);
+      spannerCellsList = document.getElementsByClassName('spannerTabCell' + tableNumber + tableColumnNumber);
+
+      newColumnName = document.getElementById('columnNameText' + tableNumber + tableColumnNumber + tableColumnNumber).value;
+      originalColumnName = schemaConversionObj.SpSchema[src_table_name[tableNumber]].ColNames[tableColumnNumber];
+      
+      updatedColsData.UpdateCols[originalColumnName] = {};
+      updatedColsData.UpdateCols[originalColumnName]['Removed'] = false;
+      if (newColumnName == originalColumnName) {
+        updatedColsData.UpdateCols[originalColumnName]['Rename'] = '';
+      }
+      else {
+        updatedColsData.UpdateCols[originalColumnName]['Rename'] = newColumnName;
+      }
+      updatedColsData.UpdateCols[originalColumnName]['NotNull'] = '';
+      updatedColsData.UpdateCols[originalColumnName]['PK'] = '';
+
+      saveSpannerColumnName();
+
+      updatedColsData.UpdateCols[originalColumnName]['ToType'] = document.getElementById('dataType' + tableNumber + tableColumnNumber + tableColumnNumber).value;
+      
+      saveSpannerConstraints();
+
+      if (!($(this).find("input[type=checkbox]").is(":checked"))) {
+        updatedColsData.UpdateCols[originalColumnName]['Removed'] = true;
+      }
+
+      mySelect = new vanillaSelectBox('#spConstraint' + tableNumber + tableColumnNumber, {
+        placeHolder: spPlaceholder[tableNumber][tableColumnNumber] + " constraints selected",
+        maxWidth: 500,
+        maxHeight: 300
+      });
+    }
+  })
+
+  $(tableId).each(function (index) {
```

Comment:
index isn't used here in the below function. you can safely remove.

Reply:
index is removed